asllib/Interpreter.ml

```diff
@@ -38,9 +38,9 @@ let _dbg = false
 module type S = sig
   module B : Backend.S
 
-  val run_env : (AST.identifier * B.value) list -> B.ast -> B.value B.m
-  val run : B.ast -> B.value B.m
-  val run_typed : B.ast -> StaticEnv.env -> B.value B.m
+  val run_env : (AST.identifier * B.value) list -> AST.t -> B.value B.m
```

Comment:
Out of curiosity: why did you move from B.ast to AST.t?
I understand you're dropping the primitive types, but why?